Source/Scene/GroundPolygon.js

```diff
@@ -0,0 +1,424 @@
+
+            for (j = 0; j < wallLength; ++j) {
+                ibIndices[index++] = wallIndices[j] + offset;
+            }
+            offset += wall.attributes.position.values.length / 3;
+        }
+
+        var positionBuffer = context.createVertexBuffer(vbPositions, BufferUsage.STATIC_DRAW);
+        var normalBuffer = context.createVertexBuffer(vbNormals, BufferUsage.STATIC_DRAW);
+
+        var indexDatatype = (ibIndices.BYTES_PER_ELEMENT === 2) ?  IndexDatatype.UNSIGNED_SHORT : IndexDatatype.UNSIGNED_INT;
+        var indexBuffer = context.createIndexBuffer(ibIndices, BufferUsage.STATIC_DRAW, indexDatatype);
+
+        var attributes = [{
+            index                  : attributeLocations.positionHigh,
+            vertexBuffer           : positionBuffer,
+            componentsPerAttribute : 3,
+            componentDatatype      : ComponentDatatype.FLOAT,
+            offsetInBytes          : 0,
+            strideInBytes          : ComponentDatatype.getSizeInBytes(ComponentDatatype.FLOAT) * 3 * 2
+        }, {
+            index                  : attributeLocations.positionLow,
+            vertexBuffer           : positionBuffer,
+            componentsPerAttribute : 3,
+            componentDatatype      : ComponentDatatype.FLOAT,
+            offsetInBytes          : ComponentDatatype.getSizeInBytes(ComponentDatatype.FLOAT) * 3,
+            strideInBytes          : ComponentDatatype.getSizeInBytes(ComponentDatatype.FLOAT) * 3 * 2
+        }, {
+            index                  : attributeLocations.normal,
+            vertexBuffer           : normalBuffer,
+            componentsPerAttribute : 3,
+            componentDatatype      : ComponentDatatype.FLOAT
+        }];
+
+        polygon._va = context.createVertexArray(attributes, indexBuffer);
+
+        polygon._boundingSphere = BoundingSphere.fromPoints(outerRing);
+        polygon._boundingSphere.radius = upDelta; // TODO: Correct bounding sphere
+    }
+
+    GroundPolygon.prototype.update = function(context, frameState, commandList) {
+        if (!defined(this._va)) {
+            createShadowVolume(this, context);
+        }
+
+        if (!defined(this._sp)) {
+            this._sp = context.createShaderProgram(ShadowVolumeVS, ShadowVolumeFS, attributeLocations);
+        }
+
+        if (!defined(this._stencilPassCommand)) {
+            var stencilPassRenderState = context.createRenderState({
+                colorMask : {
+                    red : false,
+                    green : false,
+                    blue : false,
+                    alpha : false
+                },
+                stencilTest : {
+                    enabled : true,
+                    frontFunction : StencilFunction.ALWAYS,
+                    frontOperation : {
+                        fail : StencilOperation.KEEP,
+                        zFail : StencilOperation.KEEP,
+                        zPass : StencilOperation.INCREMENT_WRAP
```

Comment:
Do you mind changing this and `DECREMENT_WRAP` to match WebGL?  This is a private change so it is not breaking.

When I first started the renderer, I spelled everything out then I realized it was error prone to be inconsistent with WebGL, especially in JavaScript.